crates/forge/src/init.rs

```diff
@@ -16,6 +16,48 @@ static TEMPLATE: Dir = include_dir!("starknet_forge_template");
 const DEFAULT_ASSERT_MACROS: Version = Version::new(0, 1, 0);
 const MINIMAL_SCARB_FOR_CORRESPONDING_ASSERT_MACROS: Version = Version::new(2, 8, 0);
 
+const DEPLOY_SCRIPT_NAME: &str = "deploy.cairo";
+
+const DEPLOY_SCRIPT_TEMPLATE: &str = r#"use sncast_std::{
+    declare, declare::DeclareResult, 
+    deploy, deploy::DeployResult,
+    get_nonce, 
+    get_balance,
+    ContractClassType,
+};
+
+// Default deployment script for {{ PROJECT_NAME }}
+fn main() {
+    // Getting the current nonce for deployment
+    let nonce = get_nonce().unwrap();
+    println!("Current nonce: {}", nonce);
+    
+    // Checkinng account balance before deployment
+    let balance = get_balance().unwrap();
+    println!("Account balance: {}", balance);
+
+    // Declaring the contract using project name placeholder
+    // This will be replaced with the actual project name during initialization
+    let declare_result = declare("{{ PROJECT_NAME }}", ContractClassType::V2).expect('Declaration failed');
+    println!("Contract declared with class hash: {}", declare_result.class_hash);
+    
+    // Preparing to deploy your contract
+    let constructor_calldata = array![];    // empty array, incase of contracts without a constructor. You cad add constructor arguments if needed
```

Comment:
```suggestion
    let constructor_calldata = array![];    // empty array, in case of contracts without a constructor. You can add constructor arguments if needed
```